packages/typescript-checker/src/typescript-compiler.ts

```diff
@@ -0,0 +1,251 @@
+          if (file.fileName.includes('.d.ts')) return false;
+          if (file.fileName.includes('node_modules')) return false;
+          return true;
+        }
+
+        return program;
+      },
+      (error) => {
+        this.currentErrors.push(error);
+      },
+      (status) => {
+        this.log.debug(status.messageText.toString());
+      },
+      (summary) => {
+        summary.code !== FILE_CHANGE_DETECTED_DIAGNOSTIC_CODE && this.currentTask.resolve();
+      }
+    );
+
+    const compiler = ts.createSolutionBuilderWithWatch(host, [this.tsconfigFile], {});
+    compiler.build();
+    return await this.check([]);
+  }
+
+  public async check(mutants: Mutant[]): Promise<ts.Diagnostic[]> {
+    this.lastMutants.forEach((mutant) => {
+      const file = this.fs.getFile(mutant.fileName);
+      file?.resetMutant();
+    });
+    mutants.forEach((mutant) => {
+      const file = this.fs.getFile(mutant.fileName);
+      file?.mutate(mutant);
+    });
+    await this.currentTask.promise;
+    const errors = this.currentErrors;
+    this.currentTask = new Task();
+    this.currentErrors = [];
+    this.lastMutants = mutants;
+    return errors;
+  }
+
+  public getFileRelationsAsNodes(): Map<string, Node> {
+    if (!this.nodes.size) {
+      // create nodes
+      for (const [fileName] of this.sourceFiles) {
+        const backslashFileName = toBackSlashFileName(fileName);
+        const node = new Node(backslashFileName, [], []);
+        this.nodes.set(backslashFileName, node);
+      }
+
+      // set childs
+      for (const [fileName, file] of this.sourceFiles) {
+        const node = this.nodes.get(toBackSlashFileName(fileName));
+        if (node == null) {
+          throw new Error('todo');
+        }
+
+        const importFileNames = [...file.imports];
+        // todo fix !
+        node.childs = importFileNames.map((importName) => this.nodes.get(toBackSlashFileName(importName))!).filter((n) => n != undefined);
+      }
+
+      for (const [, node] of this.nodes) {
+        node.parents = [];
+        for (const [_, n] of this.nodes) {
+          if (n.childs.includes(node)) {
+            node.parents.push(n);
+          }
+        }
+      }
+    }
+
+    return this.nodes;
+  }
+
+  private resolveFilename(fileName: string): string[] {
+    if (!fileName.includes('.d.ts')) return [fileName];
+
+    const file = this.fs.getFile(fileName);
+    if (!file) throw new Error(`Could not find ${fileName}`);
+    const sourceMappingURL = this.getSourceMappingURL(file.content);
+
+    if (!sourceMappingURL) return [fileName];
+
+    const sourceMapFileName = path.resolve(fileName, '..', sourceMappingURL);
+    const sourceMap = this.fs.getFile(sourceMapFileName);
+    if (!sourceMap) throw new Error(`Could not find ${sourceMapFileName}`);
+
+    const content = JSON.parse(sourceMap.content);
+
+    return content.sources.map((sourcePath: string) => toPosixFileName(path.resolve(sourceMapFileName, '..', sourcePath)));
+  }
```

Comment:
What is this method?

Reply:
This method resolves the original filename(s) based on a .d.ts file.

Reply:
I saw that :). Let me clarify:
1. Why do we need to resolve the original TS file name? 
2. Shouldn't we call it `resolveTSInputFile` instead?
3. Does this work when [inlineSourceMap](https://www.typescriptlang.org/tsconfig#inlineSourceMap), [inlineSources](https://www.typescriptlang.org/tsconfig#inlineSources), [mapRoute](https://www.typescriptlang.org/tsconfig#mapRoot) or [sourceRoute](https://www.typescriptlang.org/tsconfig#sourceRoot) are defined ?

Reply:
1. We need the source filename to complete the dependency graph used by the grouping algorithm. If we don't use the source filename we can not identify the relation between projects/files.
2. I would only add the word `name` so it is clear that it is about file names. Do you agree with `resolveTSInputFileName`?
3. Nope. Should we try to implement it or could we just remove/change those options without sacrifices in the `tsconfig-helpers.ts` file?